src/ribbon/buttonbar.cpp

```diff
@@ -92,30 +81,39 @@ class wxRibbonButtonBarButtonBase
         barButtonSmallImageListPos = -1;
     }
 
+
     void SetBitmaps(wxRibbonBar* ribbon,
                     wxSize bitmap_size_large,
                     wxSize bitmap_size_small,
-                    wxBitmap bitmap_large,
-                    wxBitmap bitmap_large_disabled,
-                    wxBitmap bitmap_small,
-                    wxBitmap bitmap_small_disabled)
+                    const wxBitmapBundle &bundle_large,
+                    const wxBitmapBundle &bundle_large_disabled,
+                    const wxBitmapBundle &bundle_small,
+                    const wxBitmapBundle &bundle_small_disabled)
     {
+        double scale = ribbon->GetDPIScaleFactor();
+        wxSize szLarge = bitmap_size_large * scale;
+        wxSize szSmall = bitmap_size_small * scale;
+
+        wxBitmap bitmap_large = (bundle_large.IsOk()) ?
```

Comment:
We don't need `IsOk()` tests here, calling `GetBitmap()` on a null bundle returns a null bitmap anyhow and omitting these checks would make the code simpler and shorter.

Reply:
OK, I will rethink this.